File: subprojects/common/ddo-dal/src/main/java/io/truthencode/ddo/dal/repositories/CategoryRepository.java
package io.truthencode.ddo.dal.repositories;


import io.quarkus.hibernate.reactive.panache.PanacheRepository;
import io.truthencode.ddo.dal.entity.Category;
import jakarta.enterprise.context.ApplicationScoped;

/**
 * Repository for managing Category entities using Quarkus Hibernate Reactive Panache.
 *
 * This repository provides standard CRUD operations for Category entities
 * within the application's data access layer.
 */
@ApplicationScoped
public class CategoryRepository implements PanacheRepository<Category> {
}

File: subprojects/common/ddo-dal/src/main/kotlin/io/truthencode/ddo/dal/repository/CategoryRepository.kt
package io.truthencode.ddo.dal.repository

import io.quarkus.hibernate.orm.panache.kotlin.PanacheRepository
import io.truthencode.ddo.dal.entity.Category
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class CategoryRepository : PanacheRepository<Category> {
    fun findByName(name: String) = find("name", name).firstResult()

    fun all() = listAll()
}